nimbess-cni.go

```diff
@@ -0,0 +1,314 @@
+// Copyright (c) 2019 Red Hat and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Main package for the Nimbess CNI plugin
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"os"
+	"time"
+
+	"github.com/containernetworking/cni/pkg/skel"
+	"github.com/containernetworking/cni/pkg/types"
+	"github.com/containernetworking/cni/pkg/version"
+	"google.golang.org/grpc"
+
+	cnitypes "github.com/containernetworking/cni/pkg/types/current"
+	bv "github.com/containernetworking/plugins/pkg/utils/buildversion"
+	cninimbess "github.com/nimbess/nimbess-agent/pkg/proto/cni"
+	log "github.com/sirupsen/logrus"
+)
+
+const defaultLogFile = "/tmp/nimbess-cni.log"
+
+// NimbessConfig is whatever you expect your configuration json to be. This is whatever
+// is passed in on stdin. Your plugin may wish to expose its functionality via
+// runtime args, see CONVENTIONS.md in the CNI spec.
+type nimbessConfig struct {
+	// common CNI config
```

Comment:
what does common mean? Seems like there is only one config and not something where you would have a common section then followed by different sections based on the context.